src/main/java/seedu/address/storage/ClassStorage.java

```diff
@@ -0,0 +1,122 @@
+package seedu.address.storage;
+
+import java.time.LocalDate;
+import java.time.LocalTime;
+import java.util.ArrayList;
+import java.util.HashMap;
+import java.util.List;
+
+import javafx.collections.ObservableList;
+import seedu.address.logic.commands.EditCommand;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.ReadOnlyAddressBook;
+import seedu.address.model.person.Class;
+import seedu.address.model.person.Person;
+
+/**
+ * Manages storage of AddressBook class data.
+ */
+public class ClassStorage {
+
+    private static HashMap<LocalDate, List<Person>> classes;
+    private ReadOnlyAddressBook addressBook;
+
+    /**
+     * Constructs a {@code ClassStorage} with the given addressBook.
+     *
+     * @param addressBook ReadOnlyAddressBook object.
+     */
+    public ClassStorage(ReadOnlyAddressBook addressBook) {
+        this.addressBook = addressBook;
+        this.classes = initialiseClass();
+    }
+
+    /**
+     * Initialises HashMap classes field.
+     *
+     * @return HashMap object.
+     */
+    public HashMap<LocalDate, List<Person>> initialiseClass() {
+        HashMap<LocalDate, List<Person>> map = new HashMap<>();
+        ObservableList<Person> listOfPersons = addressBook.getPersonList();
+        for (int i = 0; i < listOfPersons.size(); i++) {
+            Person person = listOfPersons.get(i);
+            Class classOfPerson = person.getAClass();
+            if (!classOfPerson.classDateTime.equals("")) {
+                if (!map.containsKey(classOfPerson.date)) {
+                    List<Person> ls = new ArrayList<>();
```

Comment:
Might want to change `ls` name to something clearer. Also it would be better if the generics could be enforced, ie: `List<Person> ls = new ArrayList<Person>();`